src/components/List/List.tsx

```diff
@@ -69,17 +69,21 @@ const List: VibeComponent<ListProps> & {
     },
     ref
   ) => {
-    const overrideId = useListId(id);
     const componentRef = useRef(null);
+    const mergedRef = useMergeRef(ref, componentRef);
+    const childrenRefs: React.MutableRefObject<HTMLElement[]> = useRef([]);
+
+    const overrideId = useListId(id);
     const [focusIndex, setFocusIndex] = useState(0);
-    const mergedRef = useMergeRefs({ refs: [ref, componentRef] });
     const Component = component;
-    const childrenRefs: React.MutableRefObject<HTMLElement[]> = useRef([]);
 
-    const updateFocusedItem = useCallback((id: string) => {
-      setFocusIndex(getListItemIndexById(childrenRefs, id));
-      componentRef?.current?.setAttribute("aria-activedescendant", id);
-    }, []);
+    const updateFocusedItem = useCallback(
+      (id: string) => {
+        setFocusIndex(getListItemIndexById(childrenRefs, id));
+        mergedRef?.current?.setAttribute("aria-activedescendant", id);
+      },
+      [mergedRef]
```

Comment:
componentRef wasn't a dependency, need to make sure no side effect happens (compare two behaviors in the story page?)